process/txcost/transactionCostProcessor.go

```diff
@@ -0,0 +1,175 @@
+	"net/http"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go-logger/check"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+	"github.com/ElrondNetwork/elrond-proxy-go/process"
+)
+
+// TransactionCostPath defines the transaction's cost path of the node
+const TransactionCostPath = "/transaction/cost"
+
+var log = logger.GetOrCreate("process/txcost")
+
+type transactionCostProcessor struct {
+	proc            process.Processor
+	pubKeyConverter core.PubkeyConverter
+	responses       []*data.ResponseTxCost
+}
+
+// NewTransactionCostProcessor will create a new instance of the transactionCostProcessor
+func NewTransactionCostProcessor(proc process.Processor, pubKeyConverter core.PubkeyConverter) (*transactionCostProcessor, error) {
+	if check.IfNil(proc) {
+		return nil, ErrNilCoreProcessor
+	}
+	if check.IfNil(pubKeyConverter) {
+		return nil, ErrNilPubKeyConverter
+	}
+
+	return &transactionCostProcessor{
+		proc:            proc,
+		pubKeyConverter: pubKeyConverter,
+		responses:       make([]*data.ResponseTxCost, 0),
+	}, nil
+}
+
+// RezolveCostRequest will resolve the transaction cost request
+func (tcp *transactionCostProcessor) RezolveCostRequest(tx *data.Transaction) (*data.TxCostResponseData, error) {
+	initialGasLimit := tx.GasLimit
+	if tx.GasLimit == 0 {
+		// TODO modify here if the max gas limit for simulate will be max gas limit per block
+		initialGasLimit = math.MaxUint64
+	}
+
+	res, err := tcp.doCostRequests(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(tcp.responses) < 2 {
+		return res, nil
+	}
+
+	for _, currentRes := range tcp.responses {
+		if currentRes.Data.RetMessage == "" {
+			continue
+		}
+
+		res.RetMessage = currentRes.Data.RetMessage
+		res.TxCost = 0
+		return res, nil
+	}
+
+	numRes := len(tcp.responses)
+	totalGas := tcp.responses[numRes-1].Data.TxCost + initialGasLimit - tcp.responses[numRes-2].Data.TxCost
+	res.TxCost = totalGas
+
+	return res, nil
+}
+
+func (tcp *transactionCostProcessor) doCostRequests(tx *data.Transaction) (*data.TxCostResponseData, error) {
+	senderShardID, receiverShardID, err := tcp.computeSenderAndReceiverShardID(tx.Sender, tx.Receiver)
+	if err != nil {
+		return nil, err
+	}
+
+	observers, err := tcp.proc.GetObservers(receiverShardID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		txCostResponse := &data.ResponseTxCost{}
+		respCode, errCall := tcp.proc.CallPostRestEndPoint(observer.Address, TransactionCostPath, tx, txCostResponse)
+		if respCode == http.StatusOK && errCall == nil {
+			return tcp.processResponse(senderShardID, receiverShardID, txCostResponse, tx)
+		}
+
+		// if observer was down (or didn't respond in time), skip to the next one
+		if respCode == http.StatusNotFound || respCode == http.StatusRequestTimeout {
+			log.LogIfError(err)
+			continue
+		}
+
+		// if the request was bad, return the error message
+		return nil, err
```

Comment:
if we have 2 observers in a shard and the first one is out of sync and doesn't have the transaction you are looking for, we will return error. maybe it would be helpful if we add a `continue` in this block so we can iterate over all observers in a shard

Reply:
Is not a problem if the observer is out of sync. The observer has to simulate the provided transaction .